examples/mobile/HomeApp/js/app.js

```diff
@@ -131,11 +131,77 @@ import Storage from "./clipping-storage.js";
 		appBar.removeInstantContainer(card);
 	}
 
-	function updateAppsList(apps) {
+	function updateAppsList(message) {
+		if (message.type === "full") {
+			return updateAppsListFull(message.data);
+		} else if (message.type === "diff") {
+			updateAppsListDiff(message.data);
+			return true;
+		} else {
+			console.warn("app.js: unsupported type of applist.");
+		}
+		return false;
+	}
+
+	function updateAppsListDiff(apps) {
+		apps.forEach(function (remoteApp) {
+			if (remoteApp.action === "add") { // add (or update if app already added)
+				const localApp = homeApp.appsList.filter(function (localApp) {
+					return remoteApp.appID === localApp.appID;
+				})[0];
+
+				delete remoteApp.action;
+
+				if (!localApp) { // add new
+					homeApp.appsList.push(remoteApp);
+				} else { // update local app
+					/**
+					 * @todo
+					 * Which properties we need update
+					 */
+					localApp.isActive = remoteApp.isActive;
```

Comment:
I see two problems here
1. when app is already on list but server sends new set of webclips, they will not be updated
2. when app is already on list but server sends isSelected property (e.g. in case of ads) - it will not be updated

Reply:
1.This method assumes that server sends diff not whole list.
2. We can create new issue.